### ISSUE TITLE
Smart custom knit doesn't work when `---` separator exists in Rmd

### ISSUE THREAD

**Issue by jooyoungseo** [REditorSupport/vscode-R#855]
When `---` separator exists in Rmd file, the smart custom knit doesn't work; instead, it falls back to `rmarkdown::render()` default.

However, when `---` is taken out, the custom knit works again.

This is critical because the `---` separator is often used in presentation output.

### Example 1: With `---` separator.

````
---
output: bookdown::gitbook
knit: bookdown::render_book
---

# Test

Para.

---

# Next
````

### Example 2: Without `---` separator.

````
---
output: bookdown::gitbook
knit: bookdown::render_book
---

# Test

Para.

# Next
````

**Comment by renkun-ken** [comment]
@ElianHugh Maybe caused by how we handle the frontmatter?

**Comment by renkun-ken** [comment]
Also, I find that live preview does not work with such document using `bookdown::render_book`:

```
processing file: main.Rmd


  |                                                                            
  |                                                                      |   0%

  |                                                                            
  |......................................................................| 100%
  ordinary text without R code




output file: /Users/ken/.vscode-R/tmp/main.knit.md


Error in relativeTo(basepath, dir) : 
  The path /Users/ken/Workspaces/test/rbook/libs/header-attrs-2.11 does not appear to be a descendant of /Users/ken/.vscode-R/tmp/
Calls: cat ... html_dependencies_as_string -> lapply -> FUN -> relativeTo

Backtrace:

1: stop("The path ", file, " does not appear to be a descendant of ", 
2: relativeTo(basepath, dir)
3: FUN(X[[i]], ...)
4: lapply(dependencies, makeDependencyRelative, basepath = output_dir, 
5: html_dependencies_as_string(dependencies, lib_dir, output_dir)

6: as_tmpfile(html_dependencies_as_string(dependencies, lib_dir, 
7: pandoc_html_extras_args(extras, self_contained, lib_dir, output_dir)
8: base(...)

9: output_format$pre_processor(front_matter, input, runtime, knit_meta, 
10: rmarkdown::render(file_path, output_format = set_html, output_file = output_file_loc, 

11: cat(lim, rmarkdown::render(file_path, output_format = set_html, 

Execution halted

[VSC-R] main.Rmd process exited with exit code 1
```

Looks like rendering to `~/.vscode-R/tmp` makes the paths inconsistent.

**Comment by renkun-ken** [comment]
Should be fixed via #856.
